### PR TITLE
EC keygen fips check

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Still in draft form..

**Comment by slontis** [comment]
The first commit should be ignored (it is included by on #24978)

Should be rebased after https://github.com/openssl/openssl/pull/24978 is merged

**Comment by slontis** [comment]
Code style updates

File: doc/man7/EVP_PKEY-EC.pod [hunk 1]
```diff
@@ -187,6 +187,28 @@ that m > k3 > k2 > k1 > 0
 
 =back
 
+The following KeyGen settable parameter is also available for the OpenSSL FIPS
```
Comment (paulidale) [inline doc/man7/EVP_PKEY-EC.pod:190]: _key generation_ instead of _KeyGen_.

There is another instance of this on line 22 which could be updated too.

**Review by paulidale — CHANGES_REQUESTED** [review]

**Comment by slontis** [comment]
Just did a rebase.

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
ping @openssl/committers

**Review by nhorman — APPROVED** [review]

**Comment by nhorman** [comment]
merged to master, thank you!

**Pull request closed without merging**